src/client.ts

```diff
@@ -165,9 +172,14 @@ export class SubscriptionClient {
   }
 
   public unsubscribe(id: number) {
+    if (!this.subscriptions[id] && !this.waitingSubscriptions[id]) {
+      this.waitingUnsubscribes[id] = true;
```

Comment:
I'm not sure what `waitingUnsubscribes` is giving us. Each `subscribe` call is going to generate a new id and I don't see this being used elsewhere.